src/app/features/public-form/public-form.module.ts

```diff
@@ -0,0 +1,108 @@
+import { NgModule } from "@angular/core";
+import { CommonModule } from "@angular/common";
+import { RouterService } from "../../core/config/dynamic-routing/router.service";
+import { ViewConfig } from "../../core/config/dynamic-routing/view-config.interface";
+import { EntityDetailsConfig } from "../../core/entity-details/EntityDetailsConfig";
+import { EntityListConfig } from "../../core/entity-list/EntityListConfig";
+import { AdminOverviewService } from "../../core/admin/admin-overview/admin-overview.service";
+import { PublicFormConfig } from "./public-form-config";
+import { AsyncComponent, ComponentRegistry } from "app/dynamic-components";
+
+/**
+ * Configure publicly accessible forms for users without login to record some data into the system.
+ */
+@NgModule({
+  declarations: [],
+  imports: [CommonModule],
+})
+export class PublicFormModule {
+  static databaseEntities = [PublicFormConfig];
+
+  constructor(
+    components: ComponentRegistry,
+    routerService: RouterService,
+    adminOverviewService: AdminOverviewService,
+  ) {
+    components.addAll(dynamicComponents);
+    routerService.addRoutes(viewConfigs);
+    adminOverviewService.menuItems.push({
+      label: $localize`:admin menu item:Manage Public Forms`,
+      link: PublicFormConfig.route,
+    });
+  }
+}
+
+const dynamicComponents: [string, AsyncComponent][] = [
+  [
+    "EditPublicFormField",
+    () =>
+      import(
+        "app/features/public-form/edit-public-form-field/edit-public-form-field.component"
+      ).then((c) => c.EditPublicFormFieldComponent),
+  ],
+];
+
+const viewConfigs: ViewConfig[] = [
+  // List View
+  {
+    _id: "view:" + PublicFormConfig.route,
+    component: "EntityList",
+    config: {
+      entityType: PublicFormConfig.ENTITY_TYPE,
+      columns: ["title", "description", "entity"],
+      filters: [{ id: "entity" }],
+    } as EntityListConfig,
+  },
+
+  // Details View
+  {
+    _id: "view:" + PublicFormConfig.route + "/:id",
+    component: "EntityDetails",
+    config: {
+      entityType: PublicFormConfig.ENTITY_TYPE,
+      panels: [
+        {
+          title: "General Setting",
+          components: [
+            {
+              component: "Form",
+              config: {
+                fieldGroups: [
+                  {
+                    fields: ["route", "title"],
+                  },
+                  {
+                    fields: ["description", "entity"],
```

Comment:
The issue during testing made me realize that admins may face problems creating new forms still because the `Config:Permissions` need to allow public role to create records of that entity ...

Ideally we would ask the user upon saving the PublicFormConfig if the Permissions shall be updated to include that rule and then edit the Config:Permissions doc automatically.
However, we don't really have a good entrypoint for this functionality when we only use the standard EntityDetails and fields :thinking: 

For v1 let's at least add a explainer text. Something like this:
```
{ id: "permissions_remark", editComponent: "EditDescriptionOnly", label: $localize:`:PublicFormConfig admin form:If you want external people filling this form without logging in, the Permission System also has to allow "public" users to create new records of this type. If you are seeing problems submitting the form, please contact your technical support team.` }
```